### PR TITLE
add support for node bundle

### PR DESCRIPTION
**Pull request by Spikef**
when webpack config target=node, the modules are store at exports.modules.

https://github.com/webpack/webpack/blob/157c457241bf46b1cb31dbb17e806702eb4a0ec5/lib/node/NodeTemplatePlugin.js#L39

### CODE REVIEW

File: test/bundles/validNodeBundleWithModulesAsArray.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+exports.ids = ["common"];
+exports.modules = {
```
Comment (th0r) [inline test/bundles/validNodeBundleWithModulesAsArray.js:2]: Filename is "validNodeBundle**WithModulesAsArray**.js" but modules are in the object. Shall we remove `WithModulesAsArray` part?

File: src/parseUtils.js [hunk 2]
```diff
@@ -25,6 +25,21 @@ function parseBundle(bundlePath) {
     ast,
     walkState,
     {
+      AssignmentExpression(node, state,) {
```
Comment (th0r) [inline src/parseUtils.js:28]:
```suggestion
      AssignmentExpression(node, state) {
```

File: src/parseUtils.js [hunk 3]
```diff
@@ -25,6 +25,21 @@ function parseBundle(bundlePath) {
...
+        if (state.locations) return;
+
+        // Modules are stored in exports.modules:
+        // exports.modules = {};
+        const {left, right} = node;
+
+        if (
+          left &&
+          left.object && left.object.name === 'exports' &&
+          left.property && left.property.name === 'modules'
```
Comment (th0r) [inline src/parseUtils.js:38]: Lets also add `&& isModulesHash(right)` check.

**Comment by th0r** [comment]
Merged, thanks!

**Comment by valscion** [comment]
This has been released in v3.8.0 :tada:

**Pull request merged**